autorag/synthesizer/render.py

```diff
@@ -11,44 +11,75 @@
 # Call the clear() method on the instance
 global_hydra.clear()
 
-@hydra.main(version_base=None, config_path="../../conf", config_name="config")
-def main(cfg: DictConfig):
-    cur_cfg = cfg.synthesizer.render
-    index_dir = cur_cfg.index_dir
-    app_description = cur_cfg.app_description
-    citation_cfg = cur_cfg.citation_cfg
-
+# initialize the index
+@st.cache_data
+def init(index_dir, enable_cite, similarity_top_k, citation_chunk_size):
     service_context = ServiceContext.from_defaults()
     # rebuild storage context
     storage_context = StorageContext.from_defaults(persist_dir=index_dir)
     # load index
     index = load_index_from_storage(storage_context, service_context=service_context)
-    if citation_cfg.enable_cite:
+    if enable_cite:
         # service_context for the synthesizer is same as service_context of the index
         query_engine = CitationQueryEngine.from_args(
             index,
-            similarity_top_k=citation_cfg.similarity_top_k,
+            similarity_top_k = similarity_top_k,
```

Comment:
Minor: remove spaces around `=`. Try some formatting tool such as [black](https://github.com/psf/black). I believe we can also add this checking to the repo

Reply:
I will reformat the whole project in the next PR, and try to make this checking required when anyone creates a PR.